phoenix-core/src/test/java/org/apache/phoenix/query/BaseTest.java

```diff
@@ -2006,4 +2015,21 @@ public void run() {
         thread.setDaemon(true);
         thread.start();
     }
+
+    /**
+     * Confirms that no storeFile has refCount leakage
+     */
+    protected synchronized static void confirmStoreRefCountLeak()
+            throws Exception {
+        if (getUtility() != null) {
+            try {
+                CompatUtil.confirmStoreRefCountLeak(
+                    getUtility().getAdmin());
+            } catch (IOException e) {
+                LOGGER.error("StoreFile refCount is leaked", e);
```

Comment:
This is a rather unexpected side effect. I am not a fan of having the freeResources logic called from a different place in the normal and the error path.

Maybe call this method freeResourcesAndCheckRefCountLeak, and 
call freeResources unconditionally from from finally here ?

Reply:
Okk I think we better let cluster shutdown happen the way it does today. We better use `finally` with catching storeFile refCount leak assertion catch so that not only we fail test but also let resources be freed up in individual test's `finally` block. I was thinking of taking care of cluster shutdown in `BaseTest` only as generic check, but now it seems better let individual test decide what they want to free up - cluster or just drop table or anything as per it's current behaviour.
The only problem with current approach is that if refCount leakage is encountered in one test and if we shutdown miniCluster in `@After`, other tests will not run anyways.

@stoty you are also fine with having `finally` in each test right?

Reply:
Sounds good, @virajjasani